dauphine-open-data-lib/src/main/java/io/github/oliviercailloux/y2018/opendata/dao/PersonDao.java

```diff
@@ -0,0 +1,100 @@
+package io.github.oliviercailloux.y2018.opendata.dao;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.function.Consumer;
+import javax.persistence.EntityManager;
+import javax.persistence.EntityTransaction;
+import javax.persistence.Query;
+import io.github.oliviercailloux.y2018.opendata.entity.Person;
+
+/**
+ * Represents class which will help to manipulate Person Objet <br />
+ * Every method rejects null argument.
+ * 
+ * @author Dauphine - GANDI Taric
+ */
+public class PersonDao extends AbstractDao<Person> {
+
+	/**
+	 * Inheritance of constructor from super class AbstractDao
+	 * 
+	 * @param entityManager
+	 * @param entityClass
+	 * @param entityName
+	 */
+	public PersonDao(EntityManager entityManager, Class<Person> entityClass, String entityName) {
+		super(entityManager, entityClass, entityName);
+	}
+
+	/**
+	 * This method help to find all records of person saved in DB
+	 */
+	@SuppressWarnings("unchecked")
+	@Override
+	public List<Person> findAll() {
+
+		Query query = entityManager.createQuery("SELECT e FROM Person e");
+		return query.getResultList();
+	}
+
+	/**
+	 * Help to remove record of person with given id
+	 */
+	@Override
+	public void remove(Long id) throws EntityDoesNotExistDaoException {
+
+		Query query = entityManager.createNativeQuery("DELETE FROM Person WHERE ID = " + id);
```

Comment:
Inélégant. Utiliser JPA plutôt qu’une requête native.

Reply:
@oliviercailloux 
Corrigé

Reply:
@oliviercailloux